libvirt/resource_libvirt_pool.go

```diff
@@ -248,9 +347,17 @@ func resourceLibvirtPoolDelete(ctx context.Context, d *schema.ResourceData, meta
 		}
 	}
 
-	err = virConn.StoragePoolDelete(pool, 0)
-	if err != nil {
-		return diag.Errorf("error deleting storage pool: %s", err)
+	poolDef, dia := newDefPoolFromLibvirt(virConn, pool)
+	if dia != nil {
+		return dia
+	}
+
+	// if the logical pool has no source device then the volume group existed before we created the pool, so we don't delete it
```

Comment:
@jimnydev can you explain more the logic behind this? why would in this case the pool not have a source?

I don't have used these myself, so I may be missing some context.